iot_config/src/main.rs

```diff
@@ -60,34 +60,36 @@ impl Daemon {
         // Install prometheus metrics exporter
         poc_metrics::start_metrics(&settings.metrics)?;
 
-        // Create database pool
-        let pool = settings.database.connect(10).await?;
-        sqlx::migrate!().run(&pool).await?;
-
         // Configure shutdown trigger
         let (shutdown_trigger, shutdown_listener) = triggered::trigger();
         tokio::spawn(async move {
             let _ = signal::ctrl_c().await;
             shutdown_trigger.trigger()
         });
 
+        // Create database pool
+        let (pool, db_join_handle) = settings.database.connect(shutdown_listener.clone()).await?;
+        sqlx::migrate!().run(&pool).await?;
+
         let listen_addr = settings.listen_addr()?;
 
-        let gateway_svc = GatewayService::new(settings).await?;
-        let route_svc = RouteService::new(settings).await?;
-        let org_svc = OrgService::new(settings, route_svc.clone_update_channel()).await?;
+        let gateway_svc = GatewayService::new(settings, pool.clone()).await?;
+        let route_svc = RouteService::new(settings, pool.clone()).await?;
+        let org_svc =
+            OrgService::new(settings, route_svc.clone_update_channel(), pool.clone()).await?;
```

Comment:
i had to refactor the startup for all of these services so this file will want a rebase

Reply:
Rebased